tools/apps/src/lib.rs

```diff
@@ -927,13 +934,15 @@ impl HttpConn for Http3Conn {
         loop {
             match self.h3_conn.poll(conn) {
                 Ok((stream_id, quiche::h3::Event::Headers { list, .. })) => {
-                    info!(
+                    error!(
                         "{} got request {:?} on stream id {}",
                         conn.trace_id(),
                         &list,
                         stream_id
                     );
 
+                    self.largest_processed_request = stream_id;
```

Comment:
Should this be somthing like `cmp::max(self.largest_processed_request, stream_id)` instead so if requests are reordered we don't decrease the value?

But also, do we need to do something to make sure reordered requests get processed correctly? Like, if we received request 4, then GOAWAY, then request 0, should we process request 0 before sending our GOAWAY?

Reply:
Also, because of how we process streams, a goaway event will be returned before any other events, so even if there are requests buffered the application will receive that one first... is this ok? Should we not return other events if we received or sent goaway?

Reply:
To clarify, this is the quiche-apps code so whatever we do is an application decision and not common to users of quiche library.

GOAWAY is asymmetric: a server that receives GOAWAY is being told not to initiate any new pushes. A client that receives GOAWAY is being told not to initiate any new requests. Both SHOULD cancel anything with a higher ID than included in GOAWAY (but quiche doesn't support that at the moment). 

quiche-server is being dumb and only generating a GOAWAY when it receives a GOAWAY. Ideally it should do something smarter but we don't have many places to gracefully close the connection. We typically close a connection when the client timeouts or there is some low-level error. We could add some special thing (drop the connection after 1000 requests? ;) ) but it adds complexity

The code here keeps track of the highest stream ID that was "processed" as soon as we extracted the headers via `poll()`. It doesn't matter if the server got the GOAWAY event before processing any headers - for the server it is actually useful because now we know the limit of pushes.

On the client side, if it receives GOAWAY before responses, it knows the limit of what the server processed and could decide to spin up another connection to deal with retrying requests, while finishing off reception of responses on the current connection.